### PR TITLE
Credentials are not sent in DeleteVolume call (#53)

### LINKED ISSUE #53: external provisioner does not pass credentials in DeleteVolume call
the provisioner only fetches the storage class when creating a new volume for a given PVC, but it does not fetch storage class when deleting an existing PV.
/assign @sbezverk

### PR DESCRIPTION
**Pull request by sbezverk**
Closes #53

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -276,13 +274,29 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
...
 	volumeId := p.volumeHandleToId(volume.Spec.CSI.VolumeHandle)
+
 	req := csi.DeleteVolumeRequest{
 		VolumeId: volumeId,
 	}
+	// get secrets if StorageClass specifies it
+	storageClassName := volume.Spec.StorageClassName
+	if len(storageClassName) != 0 {
+		if storageClass, err := p.client.StorageV1().StorageClasses().Get(storageClassName, metav1.GetOptions{}); err == nil {
+			if secretName, ok := storageClass.Parameters[secretNameKey]; ok {
```
Comment (jsafrane) [inline pkg/controller/controller.go]: Please refactor a function that can be shared in `Provision()` and `Delete()` calls.

Comment (sbezverk) [reply pkg/controller/controller.go]: done, PTAL.

**Pull request merged**